diesel/src/mysql/connection/mod.rs

```diff
@@ -12,13 +12,15 @@ use types::HasSqlType;
 
 #[allow(missing_debug_implementations, missing_copy_implementations)]
 pub struct MysqlConnection {
-    _raw_connection: RawConnection,
+    raw_connection: RawConnection,
     transaction_manager: AnsiTransactionManager,
 }
 
 impl SimpleConnection for MysqlConnection {
-    fn batch_execute(&self, _query: &str) -> QueryResult<()> {
-        unimplemented!()
+    fn batch_execute(&self, query: &str) -> QueryResult<()> {
+        self.raw_connection.enable_multi_statements(|| {
+            self.raw_connection.execute(query)
+        })
```

Comment:
Second paragraph of the commit message is addressing that case. I don't want to go further than the naive approach until I better understand the case it's describing and can test it.